src/main/java/ch/njol/skript/expressions/ExprFurnaceTime.java

```diff
@@ -0,0 +1,227 @@
+		for (Block block : getExpr().getArray(event)) {
+			Furnace furnace = (Furnace) block.getState();
+			changer.accept(furnace);
+			furnace.update(true);
+		}
+	}
+
+	private void changeCookTime(Event event, int providedTime, ChangeMode mode) {
+		switch (mode) {
+			case SET -> changeFurnaces(event, furnace -> furnace.setCookTime((short) providedTime));
+			case DELETE -> changeFurnaces(event, furnace -> furnace.setCookTime((short) 0));
+			case REMOVE -> changeFurnaces(event, furnace -> furnace.setCookTime((short) Math2.fit(0, furnace.getCookTime() - providedTime, Integer.MAX_VALUE)));
+			case ADD -> changeFurnaces(event, furnace -> furnace.setCookTime((short) Math2.fit(0, furnace.getCookTime() + providedTime, Integer.MAX_VALUE)));
+		}
+	}
+
+	private void changeTotalCookTimeBlock(Event event, int providedTime, ChangeMode mode) {
+		switch (mode) {
+			case SET -> changeFurnaces(event, furnace -> furnace.setCookTimeTotal(providedTime));
+			case DELETE -> changeFurnaces(event, furnace -> furnace.setCookTimeTotal(0));
+			case REMOVE -> changeFurnaces(event, furnace -> furnace.setCookTimeTotal(Math2.fit(0, furnace.getCookTimeTotal() - providedTime, Integer.MAX_VALUE)));
+			case ADD -> changeFurnaces(event, furnace -> furnace.setCookTimeTotal(Math2.fit(0, furnace.getCookTimeTotal() + providedTime, Integer.MAX_VALUE)));
+		}
+	}
+
+	private void changeBurnTimeBlock(Event event, int providedTime, ChangeMode mode) {
+		switch (mode) {
+			case SET -> changeFurnaces(event, furnace -> furnace.setBurnTime((short) providedTime));
+			case DELETE -> changeFurnaces(event, furnace -> furnace.setBurnTime((short) 0));
+			case REMOVE -> changeFurnaces(event, furnace -> furnace.setBurnTime((short) Math2.fit(0, furnace.getBurnTime() - providedTime, Integer.MAX_VALUE)));
+			case ADD -> changeFurnaces(event, furnace -> furnace.setBurnTime((short) Math2.fit(0, furnace.getBurnTime() + providedTime, Integer.MAX_VALUE)));
+		}
+	}
+
+	private void changeTotalCookTimeEvent(FurnaceStartSmeltEvent startSmeltEvent, ChangeMode mode, int providedTime) {
+		switch (mode) {
+			case SET -> startSmeltEvent.setTotalCookTime(providedTime);
+			case ADD -> startSmeltEvent.setTotalCookTime(Math2.fit(0, startSmeltEvent.getTotalCookTime() + providedTime, Integer.MAX_VALUE));
+			case REMOVE -> startSmeltEvent.setTotalCookTime(Math2.fit(0, startSmeltEvent.getTotalCookTime() - providedTime, Integer.MAX_VALUE));
+			case DELETE -> startSmeltEvent.setTotalCookTime(0);
+		}
+	}
+
+	private void changeBurnTimeEvent(FurnaceBurnEvent burnEvent, ChangeMode mode, int providedTime) {
+		switch (mode) {
+			case SET -> burnEvent.setBurnTime(providedTime);
+			case ADD -> burnEvent.setBurnTime(Math2.fit(0, burnEvent.getBurnTime() + providedTime, Integer.MAX_VALUE));
+			case REMOVE -> burnEvent.setBurnTime(Math2.fit(0, burnEvent.getBurnTime() - providedTime, Integer.MAX_VALUE));
+			case DELETE -> burnEvent.setBurnTime(0);
+		}
+	}
```

Comment:
why one for each? just accept a Consumer<Integer> as a parameter
Also why do the block/event methodds have the time/mode parameters in different orders?